code/scripting/api/objs/fireballclass.cpp

```diff
@@ -0,0 +1,126 @@
+//
+//
+
+#include "fireballclass.h"
+#include "model.h"
+#include "fireball/fireballs.h"
+
+namespace scripting {
+namespace api {
+
+
+//**********HANDLE: Fireballclass
+ADE_OBJ(l_Fireballclass, int, "fireballclass", "Fireball class handle");
+
+ADE_FUNC(__tostring, l_Fireballclass, NULL, "Fireball class name", "string", "Fireball class unique id, or an empty string if handle is invalid")
+{
+	int idx;
+	const char* s = nullptr;
+	if(!ade_get_args(L, "o|s", l_Fireballclass.Get(&idx), &s))
+		return ade_set_error(L, "s", "");
+
+	if(idx < 0 || idx >= Num_fireball_types)
+		return ade_set_error(L, "s", "");
+
+	return ade_set_args(L, "s", Fireball_info[idx].unique_id);
+}
+
+ADE_FUNC(__eq, l_Fireballclass, "fireballclass, fireballclass", "Checks if the two classes are equal", "boolean", "true if equal, false otherwise")
+{
+	int idx1,idx2;
+	if(!ade_get_args(L, "oo", l_Fireballclass.Get(&idx1), l_Fireballclass.Get(&idx2)))
+		return ade_set_error(L, "b", false);
+
+	if(idx1 < 0 || idx1 >= Num_fireball_types)
+		return ade_set_error(L, "b", false);
+
+	if(idx2 < 0 || idx2 >= Num_fireball_types)
+		return ade_set_error(L, "b", false);
+
+	return ade_set_args(L, "b", idx1 == idx2);
+}
+
+ADE_VIRTVAR(Name, l_Fireballclass, "string", "Fireball class name", "string", "Fireball class unique id, or empty string if handle is invalid")
+{
+	int idx;
+	const char* s = nullptr;
+	if(!ade_get_args(L, "o|s", l_Fireballclass.Get(&idx), &s))
+		return ade_set_error(L, "s", "");
+
+	if(idx < 0 || idx >= Num_fireball_types)
+		return ade_set_error(L, "s", "");
+
+	if(ADE_SETTING_VAR && s != nullptr) {
+		auto len = sizeof(Fireball_info[idx].unique_id);
+		strncpy(Fireball_info[idx].unique_id, s, len);
+		Fireball_info[idx].unique_id[len - 1] = 0;
+	}
+
+	return ade_set_args(L, "s", Fireball_info[idx].unique_id);
+}
+
+ADE_VIRTVAR(FireballFilename, l_Fireballclass, NULL, "Fireball class animation filename (LOD 0)", "string", "Filename, or empty string if handle is invalid")
```

Comment:
"FireballFilename" is redundant because this is the fireball class.  Just call it "Filename".